### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by barryorourke**
My first attempt at adding tests to a plugin, I think everything should be covered.

### CODE REVIEW

**Comment by majormoses** [comment]
I think I ran into this and solved this by locking some dependencies: https://github.com/sensu-plugins/sensu-plugins-dcos/blob/0.0.4/sensu-plugins-dcos.gemspec#L40-L41

**Comment by barryorourke** [comment]
It's a bit of a messy pull request but I got there in the end. Never really done much more than hack together a quick script in Ruby, so it was a bit of a learning experience.

This only tests against 2.4.1 is that we are aiming for, or do we want to include more versions?

**Comment by majormoses** [comment]
we should probably do more versions. @eheydrick do we want to tackle this now or should we just push out testing with just 2.4.1?

File: test/integration/default/serverspec/check-load_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+gem_path = '/usr/local/bin'
+check_name = 'check-load.rb'
+check = "#{gem_path}/#{check_name}"
+
+describe 'ruby environment' do
+  it_behaves_like 'ruby checks', check
+end
+
+describe file(check) do
```
Comment (majormoses) [inline test/integration/default/serverspec/check-load_spec.rb]: probably best to put all the common checks into something like a spec helper that can be called, here is an example: https://github.com/sensu-plugins/sensu-plugins-dcos/blob/0.1.1/test/integration/helpers/serverspec/shared_spec.rb

https://github.com/sensu-plugins/sensu-plugins-dcos/blob/0.1.1/test/integration/default/serverspec/check-dcos-component-health.rb_spec.rb#L10-L12

**Comment by eheydrick** [comment]
I'd prefer to have testing on all ruby versions. Wondering if it would be possible to derive the kitchen suite version from the ruby version being tested in travis.

**Comment by majormoses** [comment]
I went down that route originally but ran into issues that using the docker provisioner in travis you could not mount volumes.

**Comment by majormoses** [comment]
We could maybe try making everything a shared tests then for each suite we just have some pretty bare includes and that's about it.

**Comment by majormoses** [comment]
@barryorourke I figured out how to pretty easily tests all the versions in dcos you can check it out here: https://github.com/sensu-plugins/sensu-plugins-dcos/tree/0.3.0 and the PR with the changes: https://github.com/sensu-plugins/sensu-plugins-dcos/pull/10

**Comment by barryorourke** [comment]
This should be ready to go now.

File: sensu-plugins-load-checks.gemspec [hunk 2]
```diff
@@ -25,7 +25,7 @@ Gem::Specification.new do |s|
   s.platform               = Gem::Platform::RUBY
   s.post_install_message   = 'You can use the embedded Ruby by setting EMBEDDED_RUBY=true in /etc/default/sensu'
   s.require_paths          = ['lib']
-  s.required_ruby_version  = '>= 2.0.0'
+  s.required_ruby_version  = '>= 2.1.0'
```
Comment (majormoses) [inline sensu-plugins-load-checks.gemspec:28]: makes it a breaking change (which is OK as long as it called out appropriately in CHANGELOG) and versioned appropriately on release.

**Review by majormoses — CHANGES_REQUESTED** [review]
Looks great just need a changelog entry to detail ruby 2.0 removal. See [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md#the-breaking-change) for information on how to inform on breaking changes.

**Review by majormoses — APPROVED** [review]

**Comment by majormoses** [comment]
Will get a release out as soon as travis stops having issues: https://www.traviscistatus.com/incidents/p7sw6trycn41

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-load-checks/versions/4.0.0

**Pull request merged**